pkg/cli/cvd.go

```diff
@@ -86,17 +91,65 @@ func newCVDCommand(cfgFlags *configFlags) *cobra.Command {
 	return cvd
 }
 
-func runCreateCVDCommand(flags *createCVDFlags, c *cobra.Command, _ []string) error {
+func runCreateCVDCommand(opts *cvdCommandOptions, flags *createCVDFlags, c *cobra.Command, _ []string) error {
 	apiClient, err := buildAPIClient(flags.subCommandFlags, c)
 	if err != nil {
 		return err
 	}
 	if flags.LocalImage {
-		return errors.New("Not implemented yet")
+		return createCVDFromLocalBuild(opts, apiClient, c, flags)
 	}
 	return createCVDFromAndroidCI(apiClient, c, flags)
 }
 
+const RequiredImagesFilename = "device/google/cuttlefish/required_images"
+
+type MissingEnvVarErr string
+
+func (s MissingEnvVarErr) Error() string {
+	return fmt.Sprintf("Missing environment variable: %q", string(s))
+}
+
+type MissingRequiredImagesFileErr string
+
+func (s MissingRequiredImagesFileErr) Error() string {
+	return fmt.Sprintf("Required images file not found at %q", string(s))
+}
+
+const (
+	AndroidBuildTopVarName   = "ANDROID_BUILD_TOP"
+	AndroidProductOutVarName = "ANDROID_PRODUCT_OUT"
+	AndroidHostOutVarName    = "ANDROID_HOST_OUT"
+)
+
+func createCVDFromLocalBuild(
```

Comment:
nit: You could rewrite this function as
```
func createCVDFromLocalBuild(...) {
    // dirs is a struct with 3 string fields
    dirs, err := FileLocationsFromEnv()
    if err != nil {...}

    files, err := ListRequiredFilenames(dirs)
    if err != nil {...}

    if err := apiClient.UploadFiles(..., files); err != nil {...}
    return nil
}
```
That way you could test each of the 3 simpler functions on their own instead of the larger and more complicated `createCVDFromLocalBuild` one.

Reply:
definitely way more readable, thanks. done!